### ISSUE TITLE
Discard unfinished spans when capturing transaction

### ISSUE THREAD

**Issue by philipphofmann** [getsentry/sentry-cocoa#1057]
### Environment

Sentry SaaS (sentry.io)

sentry-cocoa 7.0.0-beta.0

### Steps to Reproduce

```swift
let transaction = SentrySDK.startTransaction(name: "Some Transaction", operation: "Some Operation")

let span1 = transaction.startChild(operation: "Child 1")
span1.finish(status: SentrySpanStatus.invalidArgument)

let span2 = transaction.startChild(operation: "Child 2")
        
transaction.finish()
```

### Expected Result

The transaction should show up in Sentry without span2.

### Actual Result

The transaction doesn't show up in Sentry.

**Comment by philipphofmann** [comment]
We can solve this by removing unfinished spans when calling captureTransaction in the client. We can use the Java SDK as a reference, see [SentryClientTest](https://github.com/getsentry/sentry-java/blob/21dd1665c9feac7e245bc123ceb18c3bb4e8b08b/sentry/src/test/java/io/sentry/SentryClientTest.kt#L766-L783) and [SentryClient](https://github.com/getsentry/sentry-java/blob/838067f920a4b1bd82ee7307da73ebdfa282d62c/sentry/src/main/java/io/sentry/SentryClient.java#L436-L448).

**Comment by brustolin** [comment]
Issue solved in PR #1066
